File: clients/typescript/src/models/Tag.ts
import type { TagFlag } from "./TagFlag.js";

/**
 * A free-form label attached to one or more {@link MindObject}s.
 * Tags have no separate identifier — they are referenced by name.
 */
export interface Tag {
  name: string;
  count: number;
  flags: TagFlag;
  /** ISO 8601 UTC timestamp of the last add/remove for this tag. */
  modified: string;
}

File: clients/csharp/src/Models/Tag.cs
using System.Text.Json.Serialization;

namespace MyMind.Models;

public class Tag
{
    [JsonPropertyName("id")]
    public required string Id { get; init; }

    [JsonPropertyName("name")]
    public required string Name { get; init; }

    [JsonPropertyName("count")]
    public required int Count { get; init; }

    [JsonPropertyName("flags")]
    public required TagFlag Flags { get; init; }

    [JsonPropertyName("modified")]
    public required DateTime Modified { get; init; }
}
